test/13_arraycommon.jl

```diff
@@ -975,9 +1019,28 @@ end
 
 @testset "broadcast" for x in (CategoricalArray(1:3),
                                CategoricalArray{Union{Int,Missing}}(1:3),
-                               CategoricalArray(["a", "b", "c"]))
+                               CategoricalArray(["a", "b", "c"]),
+                               CategoricalArray{Union{String,Missing}}(["b", missing, "c"]))
+    y = identity.(x)
+    @test x ≅ y
+    @test x !== y
+    @test y isa CategoricalArray
+
+    y = broadcast(v->v, x)
+    @test x ≅ y
+    @test x !== y
+    @test y isa CategoricalArray
+
+    y = broadcast(v->x[1], x)
+    @test x !== y
+    @test y isa CategoricalArray
+
+    y = broadcast(v->1, x)
+    @test y == [1, 1, 1]
+    @test y isa Vector{Int}
+
     x[1:2] .= x[3]
-    @test x == fill(get(x[3]), 3)
+    @test x == fill(x[3], 3)
```

Comment:
does `fill` now produce `CategoricalArray` (it should 😄)? Is this tested (this test does not catch this).

Reply:
No, it's just that it's not needed here. We should fix `fill` separately...